### PR TITLE
Pipeline job to create unstable Kyma CLI upon Reconciler changes

### PR DESCRIPTION
**Pull request by pxsalehi**
**Description**
Create CI pipeline to generate nightly CLI builds using the latest reconciler

Changes proposed in this pull request:
- A nightly pipeline job to create unstable Kyma builds

**Related issue(s)**

Closes https://github.com/kyma-incubator/reconciler/issues/283
Requires https://github.com/kyma-project/cli/pull/1099 to be merged first.

### CODE REVIEW

File: prow/scripts/reconciler-publish-unstable-cli.sh [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+if clitests::testSuiteExists "test-version"; then
+    clitests::execute "test-version" "${ZONE}" "cli-integration-test-${RANDOM_ID}" "$SOURCE"
+else
+    log::error "Test file 'test-version.sh' not found"
+fi
+if clitests::testSuiteExists "test-function"; then
+    clitests::execute "test-function" "${ZONE}" "cli-integration-test-${RANDOM_ID}" "$SOURCE"
+else
+    log::error "Test file 'test-function.sh' not found"
+fi
```
Comment (pxsalehi) [inline prow/scripts/reconciler-publish-unstable-cli.sh:162]: remove

File: prow/jobs/incubator/reconciler/reconciler.yaml [hunk 2]
```diff
@@ -141,6 +141,63 @@ postsubmits: # runs on main
...
+      decoration_config:
+        grace_period: 600000000000
+        timeout: 14400000000000
+      cluster: trusted-workload
+      max_concurrency: 10
+      branches:
+        - ^master$
+        - ^main$
+      extra_refs:
+        - org: kyma-project
```
[the remark this answers is not in the document]
Comment (clebs) [reply prow/jobs/incubator/reconciler/reconciler.yaml:172]: As discussed, kyma repo is needed to run the integration tests.

[the remark this answers is not in the document]
Comment (pxsalehi) [reply prow/jobs/incubator/reconciler/reconciler.yaml:172]: This is required to run fast-integration tests inside the VM against the provisioned k3d-based Kyma cluster.

**Review by clebs — APPROVED** [review]

**Review by Halamix2 — APPROVED** [review]

**Pull request merged**